packages/shared/src/lib/abstractBoard/boardFactory.ts

```diff
@@ -123,11 +123,18 @@ function createGridBoard<TIntersection extends Intersection>(
 
 export function createGraph<TIntersection extends Intersection, TColor>(
   intersections: TIntersection[],
+  startColor: TColor,
 ): Graph<TColor> {
   const adjacencyMatrix = intersections.map((intersection) =>
-    intersection.neighbours.map((_neighbour, index) => index),
+    intersection.neighbours.map((neighbour) =>
```

Comment:
I Added a simple unit test.
It's rather difficult to write a meaningful unit test about the graph boards, because its rather abstract, and the one I added is not independent of the implementation (i.e. another implementation may work just fine, but fail the test).

Reply:
Ah yeah, that's a good point, I agree implementation-independence is a good quality to strive for.  I think your comment in the code is a sufficient mitigation, but some thoughts on how to create implementation-independent graph tests in general -

When testing `createGraph`, we could build intersections from scratch instead of a factory:

```
const intersections = [Intersection(...), Intersection(...);
intersections[0].connectTo(intersections[1], true);

const graph = createGraph(intersections);

// This should hold true regardless of impl (unless we decide to re-map IDs)
expect(graph.neighbors(0)).toEqual([1]);
```

When testing factories, just assert on properties that should remain invariant.  E.g. counts and sizes:

```
expect(intersections.length).toBe(18)
// hard to assert on each intersection because order dependent, but
// maxes, mins and other aggregators should be mostly invariant
for (intersection of intersections) {
    expect(intersection.Neighbors.length).toBeGreaterThanOrEqualTo(2);
    expect(intersection.Neighbors.length).toBeLessThanOrEqualTo(4);
    // Same with positions
    expect(intersection.Position.x).toBeGreaterThan(0.0);
}
```